frontendUiTableImpl/src/main/java/com/virtuslab/gitmachete/frontend/ui/impl/table/GitMacheteRepositoryUpdateTask.java

```diff
@@ -0,0 +1,137 @@
+package com.virtuslab.gitmachete.frontend.ui.impl.table;
+
+import static com.intellij.openapi.application.ModalityState.NON_MODAL;
+import static com.virtuslab.gitmachete.frontend.ui.impl.table.GitPathUtils.getGitDirectoryPath;
+import static com.virtuslab.gitmachete.frontend.ui.impl.table.GitPathUtils.getMacheteFilePath;
+import static com.virtuslab.gitmachete.frontend.ui.impl.table.GitPathUtils.getMainDirectoryPath;
+
+import java.nio.file.Files;
+import java.nio.file.Path;
+
+import com.intellij.openapi.progress.ProgressIndicator;
+import com.intellij.openapi.progress.Task;
+import com.intellij.openapi.project.Project;
+import com.intellij.openapi.vcs.VcsNotifier;
+import com.intellij.ui.GuiUtils;
+import git4idea.repo.GitRepository;
+import io.vavr.control.Option;
+import io.vavr.control.Try;
+import org.checkerframework.checker.index.qual.Positive;
+
+import com.virtuslab.binding.RuntimeBinding;
+import com.virtuslab.branchlayout.api.BranchLayoutException;
+import com.virtuslab.branchlayout.api.IBranchLayout;
+import com.virtuslab.branchlayout.api.manager.IBranchLayoutManager;
+import com.virtuslab.branchlayout.api.manager.IBranchLayoutManagerFactory;
+import com.virtuslab.gitmachete.backend.api.IGitMacheteRepository;
+import com.virtuslab.gitmachete.backend.api.IGitMacheteRepositoryFactory;
+import com.virtuslab.gitmachete.backend.api.MacheteFileReaderException;
+import com.virtuslab.gitmachete.frontend.ui.api.table.BaseGraphTable;
+import com.virtuslab.logger.IPrefixedLambdaLogger;
+import com.virtuslab.logger.IntelliJLoggingUtils;
+import com.virtuslab.logger.PrefixedLambdaLoggerFactory;
+
+public final class GitMacheteRepositoryUpdateTask extends Task.Backgroundable {
+
+  private static final IPrefixedLambdaLogger LOG = PrefixedLambdaLoggerFactory.getLogger("frontendUiTable");
+
+  private final BaseGraphTable graphTable;
+  private final Project project;
+  private final GitRepository gitRepository;
+
+  private final IGitMacheteRepositoryFactory gitMacheteRepositoryFactory;
+  private final IBranchLayoutManagerFactory branchLayoutManagerFactory;
+
+  private GitMacheteRepositoryUpdateTask(BaseGraphTable graphTable, Project project, GitRepository gitRepository) {
+    super(project, "Updating Git Machete repository");
```

Comment:
![image](https://user-images.githubusercontent.com/19799111/81400985-9c98bc80-912e-11ea-8d4d-6c7ea32968d5.png)

Is it a conscious decision not to stick with IntelliJ here?

Reply:
Yes, to keep the name `Git Machete` itself properly capitalized